source/device/device.hpp

```diff
@@ -30,7 +30,7 @@ class IRoutingDevice {
     virtual bool add_inlink(std::shared_ptr<ILink> link) = 0;
     virtual bool add_outlink(std::shared_ptr<ILink> link) = 0;
     virtual bool update_routing_table(std::shared_ptr<IRoutingDevice> dest,
-                                      std::shared_ptr<ILink> link) = 0;
+                                      std::shared_ptr<ILink> link, int paths = 1) = 0;
```

Comment:
I think `size_t` fits better
```suggestion
                                      std::shared_ptr<ILink> link, size_t paths = 1) = 0;
```